libs/image-editor/ImageEditor/src/main/kotlin/org/wordpress/android/imageeditor/preview/PreviewImageViewModel.kt

```diff
@@ -35,14 +42,24 @@ class PreviewImageViewModel : ViewModel() {
             currentPosition = currentPosition,
             loadSuccess = true
         )
-        // TODO: handle file save on done button action to be added later
-//        val highResImageJustLoadedIntoView = newState != currentState && newState == ImageInHighResLoadSuccessUiState
-//        val imageNotLoadedIntoFile = loadIntoFile.value !is ImageLoadToFileSuccessState
-//        if (highResImageJustLoadedIntoView && imageNotLoadedIntoFile) {
-//            updateLoadIntoFileState(ImageStartLoadingToFileState(currentUrl))
-//        }
-        val currentUiState = uiState.value as UiState
-        updateUiState(currentUiState.copy(viewPagerItemsStates = newImageUiStates))
+        val newImageState = newImageUiStates[currentPosition]
+
+        val currentUiState = uiState.value?.peekContent() as UiState
+        val currentImageState = currentUiState.viewPagerItemsStates[currentPosition]
+
+        if (currentUiState.viewPagerItemsStates.hasSingleElement() && canLoadToFile(newImageState)) {
+            updateLoadIntoFileState(
+                ImageStartLoadingToFileState(
+                    imageUrl = currentImageState.data.highResImageUrl,
+                    position = currentPosition
+                )
+            )
+        }
+        updateUiState(currentUiState.copy(
+                viewPagerItemsStates = newImageUiStates,
+                editActionsEnabled = shouldEnableEditActionsForImageState(newImageState)
```

Comment:
Should we really change `shouldEnableEditActionsForImageState(newImageState)` here? Are we sure the image that has just finished loading is the selected image? AFAIU more than one image can be loading when we are for example in the middle of a swipe, right?

I think I managed to reproduce this issue - I changed the url of the second dummy image to a non-existing URL so it'd keep failing -> this is what I recorded https://cloudup.com/co9XP0q-HrR -> notice the crop button is sometimes disabled and sometimes enabled.

Reply:
Fixed here: 39b3b44